tensorboard/webapp/widgets/data_table/data_table_test.ts

```diff
@@ -456,4 +506,225 @@ describe('data table', () => {
       },
     ]);
   });
+
+  it('does not show add button when there are no selectable columns', () => {
+    const fixture = createComponent({});
+    expect(fixture.debugElement.query(By.css('.add-column-button'))).toBeNull();
+  });
+
+  it('renders column selector when + button is clicked', () => {
+    const fixture = createComponent({
+      selectabelColumns: [
+        {
+          type: ColumnHeaderType.HPARAM,
+          name: 'lr',
+          displayName: 'learning rate',
+          enabled: false,
+        },
+      ],
+    });
+    expect(
+      fixture.debugElement.query(By.directive(ColumnSelectorComponent))
+    ).toBeNull();
+    const addBtn = fixture.debugElement.query(By.css('.add-column-btn'));
+    addBtn.nativeElement.click();
+    expect(
+      fixture.debugElement.query(By.directive(ColumnSelectorComponent))
+    ).toBeDefined();
+  });
+
+  describe('context menu', () => {
+    let mockTableData: TableData[];
+    let mockHeaders: ColumnHeader[];
+    let mockSelectableColumns: ColumnHeader[];
+
+    beforeEach(() => {
+      mockHeaders = [
+        {
+          name: 'run',
+          type: ColumnHeaderType.RUN,
+          displayName: 'Run',
+          enabled: true,
+        },
+        {
+          name: 'disabled_header',
+          type: ColumnHeaderType.MAX_VALUE,
+          displayName: 'disabled',
+          enabled: false,
+        },
+        {
+          name: 'other_header',
+          type: ColumnHeaderType.HPARAM,
+          displayName: 'Display This',
+          enabled: true,
+        },
+      ];
+      mockTableData = [
+        {
+          id: 'runid',
+          run: 'run name',
+          disabled_header: 'disabled header',
+          other_header: 'other header',
+        },
+      ];
+      mockSelectableColumns = [
+        {
+          type: ColumnHeaderType.HPARAM,
+          name: 'lr',
+          displayName: 'Learning Rate',
+          enabled: false,
+        },
+      ];
+    });
+    it('renders context menu when a column cell is clicked', () => {
+      const fixture = createComponent({
+        headers: mockHeaders,
+        data: mockTableData,
+        selectabelColumns: mockSelectableColumns,
+      });
+      expect(fixture.debugElement.query(By.css('.context-menu'))).toBeNull();
+      const cell = fixture.debugElement.query(
+        By.directive(ContentCellComponent)
+      );
+      cell.nativeElement.dispatchEvent(new MouseEvent('contextmenu'));
+      fixture.detectChanges();
+
+      expect(
+        fixture.debugElement.query(By.css('.context-menu'))
+      ).not.toBeNull();
```

Comment:
I kind of remember you mentioning you do not like the truthy checks. I do not remember why. This seems like the correct spot for them.  I do not think this could ever be undefined so I guess this works just fine.

Reply:
This does seem like a valid usecase for a truthy check. I went with this approach because it mirrored the above `toBeNull` check.